src/components/ProgressAnimation/index.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Progress animation component.
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { purple, gray } from '@carbon/colors';
+
+import { getComponentNamespace } from '../../globals/namespace';
+
+const namespace = getComponentNamespace('progress-animation');
+
+function ProgressAnimation({
+  children,
+  className,
+  id,
+  filledColor,
+  unfilledColor,
+  percentage,
+  animationTimer,
+}) {
+  const progressAnimationName = `progress-animation-${id}`;
+
+  return (
+    <div className={classnames(namespace, className)}>
+      {/**
+       * If `animationTimer` is `null`, no need to insert a style tag.
+       * Otherwise, a new style tag with a unique keyframe needs to be
+       * created per component instance, because the unique progress
+       * percetange is used in the keyframe animation.
+       */}
+      {animationTimer ? (
+        <style
+          // eslint-disable-next-line react/no-danger
+          dangerouslySetInnerHTML={{
+            __html: `
+              @keyframes ${progressAnimationName} {
+                from {
+                  stroke-dashoffset: 0;
+                }
+                to {
+                  stroke-dashoffset: ${percentage};
+                }
+            `,
+          }}
+        />
+      ) : (
+        ''
```

Comment:
I changed this to a ternary instead of the this pattern...

```jsx
animationTimer && (
  <Style />
)
```

because weirdly, when I would set `animationTimer` to `0` in the storybook knobs, a `0` character was being rendered ON the component? 🤔 Bizarre...

Reply:
That makes sense as the `false` value will convert to `0` when coerced as a string

Reply:
Yes, I understand that. 😅😉 That’s why I used the ternary.